src/sdk/pynni/nni/nas/pytorch/darts/mutator.py

```diff
@@ -63,18 +63,21 @@ def sample_final(self):
                 edges_max[mutable.key] = max_val
                 result[mutable.key] = F.one_hot(index, num_classes=mutable.length).view(-1).bool()
         for mutable in self.mutables:
-            if isinstance(mutable, InputChoice) and mutable.n_chosen is not None:
-                weights = []
-                for src_key in mutable.choose_from:
-                    if src_key not in edges_max:
-                        _logger.warning("InputChoice.NO_KEY in '%s' is weighted 0 when selecting inputs.", mutable.key)
-                    weights.append(edges_max.get(src_key, 0.))
-                weights = torch.tensor(weights)  # pylint: disable=not-callable
-                _, topk_edge_indices = torch.topk(weights, mutable.n_chosen)
-                selected_multihot = []
-                for i, src_key in enumerate(mutable.choose_from):
-                    if i not in topk_edge_indices and src_key in result:
-                        result[src_key] = torch.zeros_like(result[src_key])  # clear this choice to optimize calc graph
-                    selected_multihot.append(i in topk_edge_indices)
-                result[mutable.key] = torch.tensor(selected_multihot, dtype=torch.bool, device=self.device())  # pylint: disable=not-callable
+            if isinstance(mutable, InputChoice):
+                if mutable.n_chosen is not None:
+                    weights = []
+                    for src_key in mutable.choose_from:
+                        if src_key not in edges_max:
+                            _logger.warning("InputChoice.NO_KEY in '%s' is weighted 0 when selecting inputs.", mutable.key)
+                        weights.append(edges_max.get(src_key, 0.))
+                    weights = torch.tensor(weights)  # pylint: disable=not-callable
+                    _, topk_edge_indices = torch.topk(weights, mutable.n_chosen)
+                    selected_multihot = []
+                    for i, src_key in enumerate(mutable.choose_from):
+                        if i not in topk_edge_indices and src_key in result:
+                            result[src_key] = torch.zeros_like(result[src_key])  # clear this choice to optimize calc graph
```

Comment:
what is the meaning of this line?

Reply:
If an edge is never selected, there is no need to calculate any op on this edge. This is to eliminate redundant calculation. If there is any problem with this, open an issue.

Reply:
please add comments here to explain.